src/bridges/notional/NotionalBridge.sol

```diff
@@ -0,0 +1,179 @@
+    )
+        external
+        payable
+        override(BridgeBase)
+        onlyRollup
+        returns (
+            uint256 outputValueA,
+            uint256,
+            bool
+        )
+    {
+        if (_auxData != 0) {
+            outputValueA = _enter(_inputAssetA.erc20Address, _outputAssetA.erc20Address, _inputValue);
+        } else {
+            outputValueA = _exit(_inputAssetA.erc20Address, _outputAssetA.erc20Address, _inputValue);
+        }
+        if (_outputAssetA.assetType == AztecTypes.AztecAssetType.ETH) {
+            IRollupProcessor(ROLLUP_PROCESSOR).receiveEthFromBridge{value: outputValueA}(_interactionNonce);
+        }
+    }
+
+    /**
+     * @notice Helper functon to compute the fcash amount
+     * @dev If the token is a ctoken, then we need to convert the amount to its underlying in 8 decimals
+     * @dev If the token is an underlying token, then we simply scale it to 8 decimals
+     * @param _token is the address of input asset token
+     * @param _amount is the amount of input asset token
+     * @param _underlyingToken is the address of the underlying asset of _token.
+     * If _token is an underlying token, then _underlyingToken == _token. If _token is a ctoken,
+     * then _underlyingToken will be _token's underlying token.
+     * @return underlyingAmount The amount of underlying token with 8 decimal
+     */
+    function computeUnderlyingAmount(
+        address _token,
+        address _underlyingToken,
+        uint256 _amount
+    ) public returns (uint88) {
+        uint256 underlyingDecimals = ERC20(_underlyingToken).decimals();
+        if (_token != _underlyingToken) {
+            uint256 exchangeRateCurrent = CToken(_token).exchangeRateCurrent();
+            uint256 mantissa = 10 + underlyingDecimals;
+            uint256 underlyingAmount = (_amount * exchangeRateCurrent) / (10**mantissa);
+            return uint88(underlyingAmount);
+        }
+        if (underlyingDecimals >= 8) {
+            return uint88(_amount / 10**(underlyingDecimals - 8));
+        }
+        return uint88(_amount * 10**(8 - underlyingDecimals));
+    }
+
+    /**
+     * @notice Approve the fcash contract to use the input token and mint fcash token
+     * @dev If the input token is ETH, we need to deposit it into WETH. If the token is a ctoken,
+     * we need to call mintViaAsset. Otherwise, we just need to call mintViaUnderlying.
+     * @dev Will revert if the wrapped fcash contract has not been deployed
+     * @param _inputToken The address of the input asset
+     * @param _outputToken The address of the output asset
+     * @param _amount The amount of input token
+     * @return outputAmount The amount of fcash token we actually receive
+     */
+    function _enter(
+        address _inputToken,
+        address _outputToken,
+        uint256 _amount
+    ) internal returns (uint256) {
+        IWrappedfCash fcash = IWrappedfCash(_outputToken);
+        if (_inputToken == address(0)) {
```

Comment:
I think it would be cleaner and less bug prone if you simply passed to these functions AztecAsset instead of the ERC20 address. Then you could check whether the asset is ETH by checking if assetType is equal to AztecAssetType.ETH.